kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/settings/SnapshotDownloadModal.java

```diff
@@ -180,16 +208,36 @@ private void initWiregraphDownloadButtons() {
     }
 
     /*
-     * Utils
+     * OnEvents Methods
      */
 
+    private void onSearchBoxEvent(KeyUpEvent event) {
+        TextBox searchBox = (TextBox) event.getSource();
+        String searchedPid = searchBox.getValue();
+
+        if (searchedPid == null || searchedPid.isEmpty() || searchedPid.equals("")) {
+            this.pidPanel.iterator().forEachRemaining(widget -> widget.setVisible(true));
+        } else {
+            this.pidPanel.iterator().forEachRemaining(widget -> {
+                CheckBox box = (CheckBox) widget;
+                box.setVisible(box.getText().toLowerCase().contains(searchedPid.toLowerCase()));
+            });
+        }
+
+        if (this.noPidSelectedError.isVisible()) {
+            this.noPidSelectedError.setVisible(false);
+        }
+
+        updateSelectOrRemoveAllText(checkPidsCheckboxStates());
+    }
+
```

Comment:
```suggestion
    private void onSearchBoxEvent(KeyUpEvent event) {
        TextBox searchBox = (TextBox) event.getSource();
        String searchedPid = searchBox.getValue();

        if (searchedPid == null || searchedPid.isEmpty() || searchedPid.equals("")) {
            this.pidPanel.iterator().forEachRemaining(widget -> widget.setVisible(true));
        } else {
            this.pidPanel.iterator().forEachRemaining(widget -> widget
                    .setVisible(isMatchingSearch(widget, searchedPid)));
        }

        if (this.noPidSelectedError.isVisible()) {
            this.noPidSelectedError.setVisible(false);
        }

        updateSelectOrRemoveAllText(checkPidsCheckboxStates());
    }
    
    private boolean isMatchingSearch(Widget widget, String searchedPid) {
        return ((CheckBox) widget).getText().toLowerCase().contains(searchedPid.toLowerCase());
    }

```

Could it be better like this?